app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -81,34 +87,65 @@ const isElementArray = ({ max, sliceName }: IElementDefinition): boolean =>
 
 const createElementNode = (
   elementDefinition: IElementDefinition,
-  isArrayItem?: boolean
+  params: { index?: number; parentPath?: string }
 ): ElementNode => {
+  const { index, parentPath } = params;
+  const prefixPath =
+    (parentPath || elementDefinition.path?.split(".").shift()) ?? "";
+  const suffixPath = elementDefinition.path?.split(".").slice(1).join(".");
+  const elementPath = `${prefixPath}${
+    prefixPath && suffixPath && "."
+  }${suffixPath}${index !== undefined ? `[${index}]` : ""}`;
   return {
     id: uuid(),
     name: elementDefinition.id?.split(".").pop() ?? "",
     children: [],
-    path: elementDefinition.path ?? "",
+    path: elementPath,
     definition: elementDefinition,
-    isArray:
-      undefined !== isArrayItem
-        ? !isArrayItem
-        : isElementArray(elementDefinition),
+    isArray: index !== undefined ? false : isElementArray(elementDefinition),
     isSlice: !!elementDefinition.sliceName,
     kind: getKind(elementDefinition),
     type: elementDefinition.type?.map((t) => computeType(t)).join(" | "),
   };
 };
 
-const getChildrenChoices = (
+const getNewPath = (path: string): string => {
+  if (path.endsWith(`]`)) {
+    const splittedPath = path.split(".");
+    const lastPathElement = splittedPath.pop()?.split("[")[0];
+    if (lastPathElement) splittedPath.push(lastPathElement);
+    return splittedPath.join(".");
+  } else {
+    return path;
+  }
+};
```

Comment:
Do we really need this function?

Reply:
No I can handle this in a constant, I thought it would be cleaner in a function

Reply:
```suggestion
const getNewPath = (path: string): string => path.split(/[[]\d+[]]/).join("");
```

Reply:
I did it but I added a $ at the end to only delete the last index